components/lib/utils/MergeProps.js

```diff
@@ -1,5 +1,22 @@
-export function mergeProps(...props) {
+import { useContext } from 'react';
+import { PrimeReactContext } from '../api/Api';
+
+export const useMergeProps = () => {
+    const context = useContext(PrimeReactContext);
+
+    return (...props) => {
+        const options = {
+            ...(context?.ptOptions?.classNameMergeFunction && { classNameMergeFunction: context.classNameMergeFunction })
+        };
+
+        return _mergeProps(props, options);
+    };
+};
+
+// *
+export function _mergeProps(props, options = {}) {
```

Comment:
@bweis why rename this `_mergeProps` won't this break backwards compatibility for users using this published method?

Reply:
I figured that this would cause either a minor bump, or be reserved for a major bump.  The `_mergeProps` function fundamentally works without any context.  Conceivably, you would want to either enforce subscribers to the API to use `useMergeProps` "hook".  Which for the record, is not a hook.  If we wanted a real "equivalent" API then `useMergeProps` should also be configured to return a custom `-mergeProps` function with the context at creation to make it actually a hook.  Rather than a simple "getter" utility.

Reply:
I totally understand I just worry about public renames of already published methods.